src/components/features/TableForm/TableForm.js

```diff
@@ -26,34 +26,33 @@ const TableForm = props => {
     if(table.id === -1){
        navigate('/', { replace: true });
      }
-  },[]);
+  },[table.id]);
 
-  const handleStatusChange = e => {
-    setStatus(e.target.value);
-    if (e.target.value === 'Cleaning' || 'Free') {
+  const handleStatusChange = value => {
+    setStatus(value);
+    if (value === 'Cleaning') {
```

Comment:
Jak wywołujesz ten sam kod dla kilku warunków, to ten wcześniejszy zapis był prawidłowy, tylko powinien wyglądać tak:
`if (value === 'Cleaning' || value === 'Free') ` 
Można też to zrobić np. za pomocą switcha:
```
switch(value) {
   case 'Cleaning':
   case 'Free':
      setPeopleAmount(value);
      break;
   default: break;
}
```

Możesz też np. podać tablicę wartości i sprawdzić czy tam istnieje:
`if (['Cleaning', 'Free'].includes(value))`
lub 
`if (['Cleaning', 'Free'].indexOf(value) !== -1)`